src/sentry/static/sentry/app/components/organizations/timeRangeSelector/utils.jsx

```diff
@@ -35,3 +38,25 @@ export function parseStatsPeriod(statsPeriod) {
     end: moment().format(DATE_TIME_FORMAT),
   };
 }
+
+/**
+ * Given a relative stats period, e.g. `1h`, return a pretty string if it
+ * is a default stats period. Otherwise if it's a valid period (can be any number
+ * followed by a single character s|m|h|d) display "Other" or "Invalid period" if invalid
+ *
+ * @param {String} relative Relative stats period
+ */
+export function getRelativeSummary(relative) {
+  const defaultRelativePeriodString = DEFAULT_RELATIVE_PERIODS[relative];
```

Comment:
Whilst I assume the output is the same it seems a bit odd to use a different logic for 4 examples in this list if any combination of positive integer + d|h|m|s are valid

Reply:
What do you suggest?

Reply:
I assume the function would return the same result if this line was omitted and we go straight to the try/catch since `parseStatsPeriod` already covers these 4 cases?

Reply:
Ah there are 3 different cases:
* if it is one of the default periods in the dropdown list, then it will return one of the defined relative periods (e.g. "Last 7 days")
* if we can parse the relative period string then we return "Other"
* otherwise "Invalid"

Reply:
Oh I see, sounds good then